websites/A/Anime4up/presence.ts

```diff
@@ -1,22 +1,161 @@
-const presence = new Presence({
-  clientId: "770030754356396052"
-});
-
-let video = {
-  duration: 0,
-  currentTime: 0,
-  paused: true
+type Functionlize<T> = {
+  [P in keyof T]: () => T[P];
 };
 
-function getTimestamps(
-  videoTime: number,
-  videoDuration: number
-): Array<number> {
-  const startTime = Date.now(),
-    endTime = Math.floor(startTime / 1000) - videoTime + videoDuration;
-  return [Math.floor(startTime / 1000), endTime];
+interface Route extends Functionlize<Partial<PresenceData>> {
+  path: RegExp;
+  playback?(): boolean;
+  run?(): PresenceData;
+}
+
+enum Settings {
+  TIMESTAMP = "timestamp",
+  BUTTONS = "buttons",
+  LOGO = "logo"
+}
+
+enum Icons {
+  PAUSED = "paused",
+  PLAYED = "played",
+  SEARCHING = "searching",
+  LOCATION = "location",
+  DISCOVERY = "discovery"
+}
+
+enum Logos {
+  LIGHT = "light-logo",
+  DARK = "dark-logo"
 }
 
+let video = { duration: 0, currentTime: 0, paused: true };
+
+const presence = new Presence({
+    clientId: "770030754356396052"
+  }),
+  startTimestamp: number = Math.floor(Date.now() / 1000),
+  router = ({ path, data }: { path: string; data: PresenceData }): Route => {
+    const routes: Route[] = [
+      { path: /^\/$/, details: () => "On Homepage" },
+      {
+        path: /^\/episode\//,
+        run: () => {
+          const timestamps = presence.getTimestamps(
+            Math.floor(video.currentTime),
+            Math.floor(video.duration)
+          );
+
+          [, data.endTimestamp] = timestamps;
+
+          if (video.paused) {
+            delete data.startTimestamp;
+            delete data.endTimestamp;
+          }
+
+          return data;
+        },
+
+        playback: () => !video.paused,
+
+        smallImageKey: () => (video.paused ? Icons.PAUSED : Icons.PLAYED),
+
+        smallImageText: () => (video.paused ? "Paused" : "Played"),
+
+        state: () =>
+          document
+            .querySelectorAll(".container")[1]
+            .textContent.slice(1, -1)
+            .split(" ")
+            .slice(0, -2)
+            .join(" "),
+
+        details: () =>
+          `Episode: ${document
+            .querySelectorAll(".container")[1]
+            .textContent.slice(1, -1)
+            .split(" ")
+            .pop()}`,
+
+        buttons: () => [
+          { label: "Watch Episode", url: location.href },
+          {
+            label: "Anime Page",
```

Comment:
```suggestion
            label: "View Page",
```

Reply:
there's a misunderstanding here, the viewed page will be the episode's page, not the anime's, so Anime Page is more accurate

Watch Episode => current location
Anime Page => the current viewed episode's anime location

Reply:
There is no misunderstanding. Then please change it to something else than `Anime Page`

Reply:
then View Anime?

Reply:
Yeah, better.